proposals/lip-Introduce_cross_chain_messages.md

```diff
@@ -0,0 +1,650 @@
+To this end, we introduce a new message format, which we call a _cross-chain message_ and define in this proposal the base schema of cross-chain messages. 
+Specifying a unified base message schema allows all chains in the Lisk ecosystem to deserialize and read cross-chain messages. 
+
+To achieve some of its basic functionalities, the Interoperability module uses four messages. 
+The first one, the _cross-chain update receipt_, is sent back to the partner chain (the chain with which the current chain is exchanging messages) whenever a cross-chain update transaction from this chain gets included. 
+The cross-chain update receipt serves to update a few properties of the interoperability store, and in general to acknowledge the inclusion of cross-chain update transactions. 
+The second one, the _channel terminated message_, is sent to chains which have been terminated. 
+This message also updates a few properties of the interoperability store to keep them synchronized between chains. 
+The third one, the _registration message_, is used when registering a chain on the Lisk mainchain. 
+The message serves as a guarantee that the correct chain ID and network ID are used when the registration transaction is sent on the sidechain.
+The last one, the _sidechain terminated message_ is created on the Lisk mainchain when a message should be routed to a terminated or inactive chain.
+This message allows other sidechains to automatically trigger the creation of the terminated sidechain account.
+
+Further motivation and rationale behind the Lisk interoperability architecture is given in the general interoperability document ["Introduce Interoperability module"][base-interoperability-LIP].
+
+
+## Rationale
+
+Messages in the Lisk ecosystem will be included in multiple chains. 
+It is therefore important that all chains can deserialize and process the base part of cross-chain messages in the same way. 
+The properties of the message contain all the information necessary for this purpose while trying to keep a minimal size.
+
+
+### Cross-chain Message Properties
+
+In the following, we list all the properties of a cross-chain message.
+
+
+#### Sending Chain ID and Receiving Chain ID
+
+Used to identify the chains exchanging the cross-chain message. 
+On the mainchain, the receiving chain ID is read to route the message to the corresponding chain outbox, as specified in [LIP "Introduce cross-chain update mechanism"][CCU-LIP]. 
+The sending chain ID is used, for example, if the message triggers an error and has to be sent back.
+
+
+#### Index
+
+When a cross-chain message is created and added to the partner chain outbox, the size of the outbox at that point is added to the message in the `index` property. 
+This allows all messages to be uniquely identified by the `(sendingChain, receivingChain, index)` tuple. 
+This property is important to identify messages and to track them throughout the ecosystem.
```

Comment:
```suggestion
The uniqueness of messages in the Lisk ecosystem is important for identifying messages with message IDs and tracking them throughout the ecosystem.
```

Reply:
As we use the message ID for tracking not the index itself.

Reply:
added